File: services/admin/src/model/book.ts
import { BookAttribute } from "./attribute";
import { Media } from "./media";

export type Book = {
    id: string;
    name: string;
    price: number;
    salePrice: number;
    pictureUrl: string;
    authorId: string;
    authorName: string;
    languageId: string;
    languageName: string;
    attributeId: string;
    attributeName: string;
    stockStatus: string;
    categories: string;
    publishDate: string;
}

export interface BookDetail {
    id: string;
    name: string;
    shortDescription: string;
    description: string;
    price: number;
    salePrice: number;
    viewCount: number;
    media: Media[];
    authorId: string;
    authorName: string;
    attributes: BookAttribute[];
    language: string;
    dimensions: string;
    publicationDate: Date;
    publisher: string;
    publicationCountry: string;
    stockStatus: string;
    totalStock: number;
    isPublic: boolean;
}

File: services/API/Domain/Book.cs
using System;
using System.Collections;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations.Schema;
using System.Linq;
using Domain.Enum;

namespace Domain
{
    public class Book
    {
        public Guid Id { get; set; }
        public string Name { get; set; }
        public string ShortDescription { get; set; }
        public string Description { get; set; }
        public bool IsPublic { get; set; }
        public DateTime CreateDate { get; set; }
        public DateTime UpdateDate { get; set; }
        public bool IsDeleted { get; set; }
        public int ViewCount { get; set; }
        public ICollection<BookCategory> Categories { get; set; }
        public ICollection<Media> Media { get; set; }
        public ICollection<BookCoupon> Coupons { get; set; }
        public ICollection<BookAttribute> Attributes { get; set; }
        public Author Author { get; set; }
        public Language Language { get; set; }
        public string Dimensions { get; set; }
        public DateTime PublicationDate { get; set; }
        public string Publisher { get; set; }
        public string PublicationCountry { get; set; }

    }
}